data/test/scenarios/overlay_text_demo.cfg

```diff
@@ -0,0 +1,270 @@
+#textdomain wesnoth-test
+
+[test]
+    name = _ "Overlay Text Demo"
+    map_file=test/maps/generic_unit_test.map
+    id = overlay_text_demo
+    turns = unlimited
+    
+    [side]
+        side=1
+        controller = human
+        suppress_end_turn_confirmation = yes
+        [leader]
+            name = _ "Alice"
+            type = Elvish Archer
+            id=alice
+            [modifications]
+                {TRAIT_QUICK}
+            [/modifications]
+        [/leader]
+    [/side]
+    [side]
+        side=2
+        controller = human
+        suppress_end_turn_confirmation = yes
+        [leader]
+            name = _ "Bob"
+            type = Orcish Grunt
+            id=bob
+            [modifications]
+                {TRAIT_QUICK}
+            [/modifications]
+        [/leader]
+    [/side]
+    [event]
+        name=preload
+        [lua]
+            code=<<
+                left_label_id = {valid = false}
+                right_label_id = left_label_id
+                top_label_id = left_label_id
+                bottom_label_id = left_label_id
```

Comment:
Wondering about the WML API that would be built over this, and how it would fit with multiple labels. I think the current `[print]` behavior of always replacing a previous label should be applied in a per-location fashion, so `[add_overlay_text]valign=bottom` would replace the previous `[add_overlay_text]valign=bottom` etc.

Or, if there isn't a WML wrapper, will it always need a prestart event like this one?
```
#define INIT_HINT_STRING_SUPPORT
    # Should be used in [event]name=prestart, or another event that triggers before the first hint.
    [lua]
        code=<<
            hint_label = {valid = false}
        >>
    [/lua]
#enddef
```

Context: I've been wondering about this as a solution to the unread walls of text in the tutorial.
```
#define SHOW_HINT_STRING TEXT
    # Similar to PRINT_STRING, but puts the text at the edge (top or bottom) of the screen.
    # Used in the tutorial for info that can be left for reading until another hint is triggered.
    #
    # There is no interaction between PRINT_STRING and SHOW_HINT_STRING - they don't cancel each
    # other's messages.
    [lua]
        code=<<
            hint_label = wesnoth.interface.add_overlay_text((...).text, {
                size = 24,
                color = {255, 255, 255},
                duration = "unlimited",
                valign = "bottom"
            })
        >>
        [args]
            text={TEXT}
        [/args]
    [/lua]
#enddef
```

Reply:
There is no WML API _yet_ but I assume there would be at some point.